test/wdio/shadow-dom-basic/cmp.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+describe('shadow-dom-basic', () => {
+  beforeEach(() => {
+    render({
+      template: () => <shadow-dom-basic-root></shadow-dom-basic-root>,
+    });
+  });
+
+  it('render', async () => {
+    await $('shadow-dom-basic-root').waitForExist();
+    const elm = document.querySelector('shadow-dom-basic-root');
+
+    if ('attachShadow' in HTMLElement.prototype) {
+      await testNativeShadow(elm);
+    } else {
+      await testPolyfilledShadow(elm);
+    }
+  });
+
+  async function testNativeShadow(elm: Element) {
+    await expect(elm.shadowRoot).toBeDefined();
+
+    const shadowDomBasic = elm.shadowRoot.lastElementChild;
+    const lightDiv = elm.shadowRoot.lastElementChild.children[0];
+
+    await expect(lightDiv.nodeName).toBe('DIV');
+    await expect(lightDiv.textContent.trim()).toBe('light');
+
+    await expect(shadowDomBasic.nodeName).toBe('SHADOW-DOM-BASIC');
+    const shadowDiv = shadowDomBasic.shadowRoot.lastElementChild.previousElementSibling;
+    await expect(shadowDiv.nodeName).toBe('DIV');
+    await expect(shadowDiv.textContent.trim()).toBe('shadow');
+
+    const shadowBG = window.getComputedStyle(shadowDiv).backgroundColor;
+    await expect(shadowBG).toBe('rgb(0, 0, 0)');
+
+    const lightBG = window.getComputedStyle(shadowDomBasic.lastElementChild).backgroundColor;
+    await expect(lightBG).toBe('rgb(255, 255, 0)');
```

Comment:
I don't think we need to `await` these `expect`s, since we're not performing any `$` or other WebdriverIO queries:
```suggestion
    expect(lightDiv.nodeName).toBe('DIV');
    expect(lightDiv.textContent.trim()).toBe('light');

    expect(shadowDomBasic.nodeName).toBe('SHADOW-DOM-BASIC');
    const shadowDiv = shadowDomBasic.shadowRoot.lastElementChild.previousElementSibling;
    expect(shadowDiv.nodeName).toBe('DIV');
    expect(shadowDiv.textContent.trim()).toBe('shadow');

    const shadowBG = window.getComputedStyle(shadowDiv).backgroundColor;
    expect(shadowBG).toBe('rgb(0, 0, 0)');

    const lightBG = window.getComputedStyle(shadowDomBasic.lastElementChild).backgroundColor;
    expect(lightBG).toBe('rgb(255, 255, 0)');
```

Reply:
I believe that's true, but with https://github.com/ionic-team/stencil/pull/5530 I think we'll start getting lint errors (even in the case of not using wdio queries)

Reply:
Hmmmm @alicewriteswrongs is that the case?

Reply:
yep! it will make us just `await` them all

but if you `await` a value which is not a promise it doesn't do anything, so I think the trade-off might be worth it if the lint rule lets us never miss an `expect` which _should_ be awaited

Reply:
ah actually I take it back! it seems the rule is smart enough to know when it's necessary

it doesn't complain about a missing `await` in

```ts
expect(2).toBe(2);
```

but it does complain about it in

```ts
    expect($('#json-foo')).toHaveText('bar');
```

Reply:
I guess it looks for `$` or at the matcher function called on the return value of `expect` or something

Reply:
looks like it checks a list of matchers which are async:

https://github.com/webdriverio/webdriverio/blob/51ac8482284ad34dea3cc899872397fb734de617/packages/eslint-plugin-wdio/src/rules/await-expect.ts#L21-L50

list here: https://github.com/webdriverio/webdriverio/blob/main/packages/eslint-plugin-wdio/src/constants.ts

Reply:
oh fancy! I can update these assertions then

Reply:
Sweet! Thanks!